[12.0] Add module stock_storage_type_putaway_abc

grindtildeath opened the pull request:
Replaces: #6 
Depends on: #8

grindtildeath committed:
Add module stock_storage_type_abc_strategy

Tonow-c2c commented on stock_storage_type_abc_strategy/models/stock_location.py:
```diff
@@ -0,0 +1,67 @@
+    display_abc_storage = fields.Boolean(
+        compute='_compute_display_abc_storage'
+    )
+    abc_storage = fields.Selection(ABC_SELECTION, required=True, default='b')
```
any help about what is the different choice signification ?

grindtildeath commented on stock_storage_type_abc_strategy/models/stock_location.py:
```diff
@@ -0,0 +1,67 @@
+            [('id', 'child_of', self.ids), ('id', '!=', self.id)],
+            order='abc_storage ASC'
+        )
+        return locations._sort_abc_locations(product.abc_storage)
```
```suggestion
        return locations._sort_abc_locations(product.abc_storage or 'b')
```

grindtildeath committed:
Rename module to stock_storage_type_putaway_abc

gurneyalex committed:
[FIX] crash when product is None

if product is None, then accessing the abc_storage will crash.

This happens when the method is called from
stock_storage_type_putaway_strategy/models/stock_storage_location_sequence.py, line 25

jgrandguillaume commented:
As it migrate to v13, I close this one
